_posts/2020-05-15-model-inversion-attacks/model_inversion_differential_privacy.Rmd

```diff
@@ -0,0 +1,740 @@
+
+But context can also be structural, such as in the scenario demonstrated in this post. For example, assume a distributed
+model, where sets lof layers run on different devices (embedded devices or mobile phones, for example). (A scenario like that
+is sometimes seen as "white-box"[@7536387], but in common understanding, white-box attacks probably presuppose some more
+insider knowledge, such as access to model architecture or even, weights. I'd therefore prefer calling this white-ish at
+most.) --- Now assume that in this context, it is possible to intercept, and interact with, a system that executes the deeper
+layers of the model. Based on that system's intermediate-level output, it is possible to perform *model inversion*[@Fred],
+that is, reconstruct the input data fed into the system.
+
+In this post, we'll demonstrate such a model inversion attack, basically porting the approach given in a
+[notebook](https://github.com/OpenMined/PySyft/blob/master/examples/tutorials/advanced/privacy_attacks/Tutorial%201%20-%20Black%20box%20model%20inversion.ipynb Sy)
+found in the [PySyft](https://github.com/OpenMined/PySyft) repository. We then experiment with different levels of
+$\epsilon$-privacy, exploring impact on reconstruction success. This second part will make use of TensorFlow Privacy,
+introduced in a [previous blog post](https://blogs.rstudio.com/ai/posts/2019-12-20-differential-privacy/).
+
+## Part 1: Model inversion in action
+
+### Example dataset: All the world's letters[^1]
+
+[^1]: Don't take *all* literally please; it's just a nice phrase.
+
+The overall process of model inversion used here is the following. With no, or scarcely any, insider knowledge about a model,
+-- but given opportunities to repeatedly query it --, I want to learn how to reconstruct unknown inputs based on just model
+outputs . Independently of original model training, this, too, is a training process; however, in general it will not involve
+the original data, as those won't be publicly available. Still, for best success, the attacker model is trained with data as
+similar as possible to the original training data assumed. Thinking of images, for example, and presupposing the popular view
+of successive layers representing successively coarse-grained features, we want that the surrogate data to share as many
+representation spaces with the real data as possible -- up to the very highest layers before final classification, ideally.
+
+If we wanted to use classical MNIST as an example, one thing we could do is
+to only use some of the digits for training the "real" model; and the rest, for training the adversary. Let's try something
+different though, something that might make the undertaking harder as well as easier at the same time. Harder, because the
+dataset features exemplars more complex than MNIST digits; easier because of the same reason: More could possibly be learned,
+by the adversary, from a complex task.
```

Comment:
```suggestion
dataset features exemplars more complex than MNIST digits; easier because of the same reason: More could possibly be learned
by the adversary from a complex task.
```